2_process/src/sites_along_waterbody.R

```diff
@@ -19,10 +18,39 @@ sites_along_waterbody <- function(sw_sites_sf, waterbody_sf, lake_waterbody = FA
       summarize(geometry = sf::st_union(geometry)) %>%
       sf::st_buffer(dist = units::set_units(250, m))
   }
-  filtered_sites <- st_join(sw_sites, waterbody_buffered, left = FALSE) %>%
+  filtered_sites <- st_join(sites_sf, waterbody_buffered, left = FALSE) %>%
     pull(site_no) %>%
     unique()
   
   return(filtered_sites)
   
 }
+
+#' @title add_stream_order
+#' @description function that joins nwis site info to nwis sw data and adds stream order col
+#' @param nwis_sw_data nwis sw data pulled from dataRetrieval::readNWISdata() or similar dataRetrieval functions 
+#' @param sites_sf sf object of sf sites and measurements
+#' @param join_site_col col to site col to join sites_sf with nwis_sw_data. defaults is site_no as that is expected to be common between both dfs
+#' @param sites_along_streamorder3 vector of site_no that are along stream order 3 streams. Output of sites_along_waterbody()    
+#' @param sites_along_lake vector of sites that are adjacent to saline lakes. Output of sites_along_waterbody
+
+add_stream_order <- function(nwis_sw_data, sites_sf, join_site_col = 'site_no',  sites_along_streamorder3, sites_along_lake){
+  
+  nwis_sw_data %>%
+    left_join(sites_sf, by = join_site_col) %>%
+    ## filtering sites that are surface level stream (all ST), lake (LK), or wetland (WE) sites
+    filter(site_tp_cd %in% c('LK','WE') | grepl('ST',site_tp_cd)) %>% 
+    ## create stream_order_category col depending on site type & match with sites stream order/lake vector targets 
+    mutate(
+      stream_order_category = case_when(
+        grepl('^ST',site_tp_cd) & site_no %in% sites_along_streamorder3 ~ 'along SO 3+',
+        site_tp_cd == 'LK' | site_no %in% sites_along_lake ~ 'along lake',
+        TRUE ~ 'not along SO 3+')) %>% 
+    ## used geometry from sites_in_watersheds_sf to get spatial info
+    st_as_sf() %>% 
+    ## grab coords 
+    mutate(lon = st_coordinates(.)[,1], lat = st_coordinates(.)[,2]) %>% 
+    ## remove geometry col
+    st_drop_geometry()  
+  
```

Comment:
This is creating a long form target with all the data with a repeating column to store site-level data. Would it be better to create site-level targets for each of the data types (sw, gw, wq) that includes all important site info like stream order, lat/lon, etc? That would be more efficient in terms of data storage, and allow us to have targets that are more flexible to downstream needs, and can be joined as needed in subsequent targets.